output-gazetteer.cpp

```diff
@@ -742,7 +740,7 @@ int output_gazetteer_t::process_relation(osmid_t id, struct member *members,
         return 0;
 
     /* get the boundary path (ways) */
-    osmid_t *xid2 = new osmid_t[member_count+1];
+    std::vector<osmid_t> xid2(member_count+1);
```

Comment:
I'm not sure we need the +1 here.

The largest access I see of xid2 is at member_count, when all members are ways.